controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -156,12 +158,12 @@ func (w *Workload) getCoreDNSInfo(ctx context.Context, clusterConfig *bootstrapv
 	}
 
 	// Handle imageRepository.
-	toImageRepository := fmt.Sprintf("%s/%s", parsedImage.Repository, parsedImage.Name)
+	toImageRepository := parsedImage.Repository
 	if clusterConfig.ImageRepository != "" {
-		toImageRepository = fmt.Sprintf("%s/%s", clusterConfig.ImageRepository, coreDNSKey)
+		toImageRepository = strings.TrimSuffix(clusterConfig.ImageRepository, "/")
```

Comment:
Do we need the trim suffix here?

Reply:
given that the image repository is provider by the user, and that down we are concatenating it with image name and tag,   I thought that removing the suffix makes the concat operation more robust (it will work with the user providing "k8s.gcr.io" but also "k8s.gcr.io/")

Reply:
Should we instead validate input and provide guidance to users?

Reply:
We can add validation and documentation as well (in another PR) but I don't think this extra caution will hurt.
But if you prefer I can remove it, it is not core to the fix we are discussing here

Reply:
Do we know how kubeadm itself handles `/` suffixes? (depending on how they are doing it, consistency might be a factor)

Reply:
AFAIK kubeadm is not doing any validation on ImageTag/ImageRepository, nor we are doing in CAPI
Also kubeadm, is not involved in immutable upgrades.

I just thought it making the operation more robust, but happy to remove it if it is blocking this PR...

Reply:
was just an idea, it's fine for me either way

Reply:
Could we open a follow-up PR to add validation?

Reply:
Adding validation could have some implications. I opened an issue to discuss them https://github.com/kubernetes-sigs/cluster-api/issues/4482